File: problems/java/ParkingLotSystem/src/model/ParkingFloor.java
package model;

import enums.VehicleSize;

import java.util.ArrayList;
import java.util.List;

/**
 * A floor in the parking lot, containing an ordered list of parking spots.
 */
public class ParkingFloor {

    private final int floorNumber;
    private final List<ParkingSpot> spots;

    /**
     * @param floorNumber the floor index (e.g. 0 for ground)
     */
    public ParkingFloor(int floorNumber) {
        this.floorNumber = floorNumber;
        this.spots = new ArrayList<>();
    }

    public int getFloorNumber() {
        return floorNumber;
    }

    public List<ParkingSpot> getSpots() {
        return spots;
    }

    public void addSpot(VehicleSize size, String id) {
        spots.add(new ParkingSpot(size, id));
    }
}

File: problems/python/ParkingLotSystem/src/model/parking_floor.py
"""A floor in the parking lot."""
from enums.vehicle_size import VehicleSize
from model.parking_spot import ParkingSpot


class ParkingFloor:
    def __init__(self, floor_number: int):
        self._floor_number = floor_number
        self._spots: list[ParkingSpot] = []

    def get_floor_number(self) -> int:
        return self._floor_number

    def get_spots(self) -> list[ParkingSpot]:
        return self._spots

    def add_spot(self, size: VehicleSize, spot_id: str) -> None:
        self._spots.append(ParkingSpot(size, spot_id))
